apps/etl-func/src/adapters/functions/messages-ingestion.ts

```diff
@@ -34,7 +33,13 @@ const messagesIngestionHandler =
     try {
       await ingestUseCase.execute(messagesMetadata);
     } catch (err) {
-      logger.error(`Error during the ingestion process`);
+      if (
+        context.retryContext?.retryCount === context.retryContext?.maxRetryCount
+      ) {
+        documents.forEach(
+          async (document) => await eventErrorRepository.push(document),
+        );
```

Comment:
Why are we pushing documents synchronously?
If we can't use a `batch` method I would do something similar

```suggestion
        Promise.all(
          documents.map((document) => eventErrorRepository.push(document)),
        );
```

Reply:
2a0ced619b90256f3803819eed888f5f8731b29e